source/env_tutorial/smac.rst

```diff
@@ -0,0 +1,368 @@
+SMAC
+~~~~~~~
+
+Overview
+==========
+
+SMAC is an environment for multi-agent collaborative reinforcement learning (MARL) on Blizzard StarCraft II. SMAC uses Blizzard StarCraft 2's machine learning API and DeepMind's PySC2 to provide a friendly interface for the interaction between agents and StarCraft 2, which is convenient for developers to observe and execute actions.
+Compared to PySC2, SMAC focuses on a decentralized micro-operation scheme, where each agent of the game is controlled by a separate RL agent.
+
+
+.. image :: ./images/smac.gif
+   :align : center
+
+Install
+========
+
+Installation method
+---------------------
+
+StarCraft 2 game and PySC2 library is needed , the installation method can refer to \ `DI-star installation <https://github.com/opendilab/DI-star>`__
+
+The installation mainly consists of two parts:
+
+1. Download the StarCraft 2 game
+For Linux system users, the installation path is \ `<https://github.com/Blizzard/s2client-proto#downloads>`__, then use the export SC2PATH=<sc2/installation/path> command to add the installation path to the environment in variable
+For Windows system users, please refer to \ `<https://starcraft2.com>`__ for installation
+
+2. Install PySC2 compatible with DI-engine
+
+.. code :: shell
+
+   git clone https://github.com/opendilab/DI-star.git
+   cd DI-star
+   pip install -e .
+
+Verify installation
+--------------------
+
+After the installation is complete, you can confirm that the environment variable is set successfully by ``echo $SC2PATH`` after the installation is successful
+
+
+Image
+------
+
+DI-engine's mirror is equipped with the framework itself and the Smac environment, which can be obtained by \ ``docker pull opendilab/ding:nightly-smac``\, or by accessing \ `docker
+hub <https://hub.docker.com/repository/docker/opendilab/ding>`__\ for more mirror
+
+
+space before transformation (original environment)
+=========================
+
+
+Observation space
+------------------
+
+- You can obtain fragmentary information such as whether each agent is alive, the remaining HP of each agent, allies or enemies within the vision range of each agent.
+
+
+Action space
+---------------
+
+- The game operation button space, generally a discrete action space of size N (N varies with the specific sub-environment), the data type is \ ``int``\ , you need to pass in python values ​​(or 0-dimensional np arrays, such as actions 3 is \``np.array(3)``\)
+
+- For each map, the action space N is generally equal to 6 + the number of enemies, such as 14 in the 3s5z map and 70 in the 2c_vs_64zg map. The specific meaning is:
+
+   - 0: NOOP
+
+   - 1: STOP
+
+   - 2: MOVE_NORTH
+
+   - 3: MOVE_SOUTH
+
+   - 4: MOVE_EAST
+
+   - 5: MOVE_WEST
+
+   - 6 - N: ATTACK ENEMY, the ID of the attacked enemy is N-6
+
+
+Bonus space
+--------
+
+- The game wins or loses, the victory is 1, and the defeat is 0, which is generally an \``int``\ value.

```

Comment:
- The game wins or loses, the victory is 1, and the defeat is 0, which is generally an \ ``int``\  value.